### DISCUSSION TITLE
Does !data?.length && !isFetching && !isLoading cover isError case?

### DISCUSSION THREAD

**Discussion by Enkratia** [reduxjs/redux-toolkit#4231] [Q&A] [answered]
Does `!data?.length && !isFetching && !isLoading` cover `isError` case?

I mean, can i use:
`!data?.length && !isFetching && !isLoading` 

Instead of:
`(!data?.length && !isLoading && !isFetching) || isError`?

**Comment by phryneas** [comment]
There is also `isUninitialized`. Maybe we add yet another one in the future?

Honestly: `isError` is `isError`. It's there to be used.

**Reply by Enkratia** [reply]
I just want to continue to show skeletons if i got the data(array), but the data is empty.
And my thought is that if i got `isError` as `true` then `!data?.length && !isFetching && !isLoading` should also to be true in the moment, 

So this `(!data?.length && !isLoading && !isFetching) || isError` 
can be simplified to this `!data?.length && !isFetching && !isLoading`;

_I will try to add more details in the future to make questions more holistic._

**Reply by phryneas** [reply, marked as answer]
You can still have `data` in the `isError - it will not disappear after an error occured, but stay available.

**Reply by Enkratia** [reply]
Ok, thank you very much.
